pkg/ottl/parser_collection_test.go

```diff
@@ -70,23 +73,36 @@ func Test_NewParserCollection_OptionError(t *testing.T) {
 
 func Test_WithParserCollectionContext(t *testing.T) {
 	ps := mockParser(t, WithPathContextNames[any]([]string{"testContext"}))
-	conv := newNopParsedStatementConverter[any]()
-	option := WithParserCollectionContext("testContext", ps, conv)
+	conv := newNopParsedStatementsConverter[any]()
+	option := WithParserCollectionContext("testContext", ps, WithStatementConverter(conv))
+
+	pc, err := NewParserCollection(componenttest.NewNopTelemetrySettings(), option)
+	require.NoError(t, err)
+
+	pw, exists := pc.contextParsers["testContext"]
+	assert.True(t, exists)
+	assert.NotNil(t, pw)
+}
+
+func Test_WithParserCollectionContextNoStatementConverter(t *testing.T) {
```

Comment:
```suggestion
func Test_ParseStatements_NoStatementConverter(t *testing.T) {
```
Can we please also add the same test for conditions?